include/tuvx/linalg/linalg.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <iostream>
+#include <random>
+#include <vector>
+
+namespace tuvx {
+namespace linalg {
+
+// tridiag matrix
+template <typename T> struct trid_mat {
+  int size;
+  std::vector<T> udiag; // upper diagonal
+  std::vector<T> ldiag; // lower diagonal
+  std::vector<T> mdiag; // main diagonal
+};
+
```

Comment:
Please add some doxygen style comments to these functions. If you're using vscode with the c++ plugins, if you type three slashes `///` it will auto create the base doxygen comments and all you have to do is fill out the details

Reply:
@AdityaDendukuri Also do `udiag`, `ldiag` and `mdiag` have the same dimension size? Theoretically `mdiag` should have one more element than the other two. We probably want to double check this and make sure that the same input can be used in LAPACK and DPLASMA directly later.

Reply:
@sjsprecious No they do not. If the matrix is `n \times x`, `udiag` and `ldiag` will be `n-1` sized vectors. I considered making them same size by padding with `0`, which might enhance the readability of the code. I will check how LAPACKE handles this.

Reply:
Awesome. Hopefully we can use the same interface for those standard math libraries and the rest part of TUV-x.

Reply:
@sjsprecious I have checked the LAPACK implementation. They do use $n-1$ sized vectors for ldiag and udiag. However, they pass in these three diagonals separately into the function rather than a combined structure like I do. Let me know if you think I should write it that way too.

Reply:
Thanks for checking it. As long as the dimension is the same, we can change the interface later as needed. No need to worry about it right now.